File: backend service/src/main/java/mutrivia/demo/backend/model/Session.java
package mutrivia.demo.backend.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Session {
    @Id @GeneratedValue(generator="system-uuid")
    @GenericGenerator(name="system-uuid", strategy = "uuid")
    private String sessionId;
    private String hostId;
    private int textDataIndex;
    private boolean isJoinable;
}

File: frontend/mutrivia-flutter/lib/models/session.dart
class Session {
  final String sessionId;
  final String hostId;
  final bool isJoinable;

  const Session({
    required this.sessionId,
    required this.hostId,
    required this.isJoinable,
  });

  factory Session.fromJson(Map<String, dynamic> json) {
    return Session(
      sessionId: json['sessionId'] as String,
      hostId: json['hostId'] as String,
      isJoinable: json['isJoinable'] as bool,
    );
  }
}